src/platforms/rcore_ios.c

```diff
@@ -0,0 +1,822 @@
+
+    // Set rendering API
+    eglBindAPI(EGL_OPENGL_ES_API);
+
+    // Create an EGL rendering context
+    platform.context = eglCreateContext(platform.device, platform.config, EGL_NO_CONTEXT, contextAttribs);
+    if (platform.context == EGL_NO_CONTEXT)
+    {
+        TRACELOG(LOG_WARNING, "DISPLAY: Failed to create EGL context");
+        return false;
+    }
+
+    // Create an EGL window surface
+    EGLint displayFormat = 0;
+
+    // EGL_NATIVE_VISUAL_ID is an attribute of the EGLConfig that is guaranteed to be accepted by ANativeWindow_setBuffersGeometry()
+    // As soon as we picked a EGLConfig, we can safely reconfigure the ANativeWindow buffers to match, using EGL_NATIVE_VISUAL_ID
+    eglGetConfigAttrib(platform.device, platform.config, EGL_NATIVE_VISUAL_ID, &displayFormat);
+
+    // eglCreateWindowSurface(platform.device, platform.config, platform.app->window, NULL);
+    // bridged cast rootViewController.view.layer; to void*
+    void* native_window = (__bridge void*)platform.viewController.view.layer;
+    platform.surface = eglCreateWindowSurface(platform.device, platform.config, native_window, NULL);
+
+    // There must be at least one frame displayed before the buffers are swapped
+    //eglSwapInterval(platform.device, 1);
+
+    if (eglMakeCurrent(platform.device, platform.surface, platform.surface, platform.context) == EGL_FALSE)
+    {
+        TRACELOG(LOG_WARNING, "DISPLAY: Failed to attach EGL rendering context to EGL surface");
+        return false;
+    }
+    else
+    {
+        CORE.Window.display.width = [[UIScreen mainScreen] nativeBounds].size.width;
+        CORE.Window.display.height = [[UIScreen mainScreen] nativeBounds].size.height;
+        if(CORE.Window.screen.width == 0){
+            CORE.Window.screen.width = [[UIScreen mainScreen] bounds].size.width;
+        }
+        if(CORE.Window.screen.height == 0){
+            CORE.Window.screen.height = [[UIScreen mainScreen] bounds].size.height;
+        }
+        
+        CORE.Window.render.width = CORE.Window.screen.width;
+        CORE.Window.render.height = CORE.Window.screen.height;
+        CORE.Window.currentFbo.width = CORE.Window.render.width;
+        CORE.Window.currentFbo.height = CORE.Window.render.height;
+
+        TRACELOG(LOG_INFO, "DISPLAY: Device initialized successfully");
+        
+        long long orientation = [[UIApplication sharedApplication] statusBarOrientation];
+        if(orientation == UIInterfaceOrientationPortrait){
+            TRACELOG(LOG_INFO,  "    > Orientation: Portrait");
+        }else if(orientation == UIInterfaceOrientationPortraitUpsideDown){
+            TRACELOG(LOG_INFO,  "    > Orientation: PortraitUpsideDown");
+        }else if(orientation == UIInterfaceOrientationLandscapeLeft){
+            TRACELOG(LOG_INFO,  "    > Orientation: LandscapeLeft");
+            swap_uint(&CORE.Window.display.width, &CORE.Window.display.height);
```

Comment:
I think it could be confusing for users to do this swap internally, I would left it to users to provide right values for width and height.

Reply:
I am going to check Android's behaviour on different orientations. I may remove this swapping.